cfme/cloud/stack.py

```diff
@@ -60,52 +66,69 @@ def edit_tags(self, tag, value):
             raise CFMEException("{} ({}) tag is not assigned!".format(tag.replace(" *", ""), value))
 
     def get_tags(self):
-        sel.force_navigate('clouds_stack', context={'stack': self})
+        navigate_to(self, 'Details')
         row = sel.elements("//*[(self::th or self::td) and normalize-space(.)={}]/../.."
                      "//td[img[contains(@src, 'smarttag')]]".format(quoteattr("My Company Tags")))
         company_tag = sel.text(row).strip()
         return company_tag
 
-    def nav_to_security_group_link(self):
-        sel.force_navigate('clouds_stack', context={'stack': self})
-        if version.current_version() <= '5.4':
-            sel.click(details_page.infoblock.element("Relationships", "Security Groups"))
-        if version.current_version() >= '5.5':
-            sel.click(details_page.infoblock.element("Relationships", "Security groups"))
-
-    def nav_to_parameters_link(self):
-        sel.force_navigate('clouds_stack', context={'stack': self})
-        sel.click(details_page.infoblock.element("Relationships", "Parameters"))
-
-    def nav_to_output_link(self):
-        sel.force_navigate('clouds_stack', context={'stack': self})
-        sel.click(details_page.infoblock.element("Relationships", "Outputs"))
-        cells = {'Key': "WebsiteURL"}
-        output_table().click_rows_by_cells(cells, "Key", True)
-
-    def nav_to_resources_link(self):
-        sel.force_navigate('clouds_stack', context={'stack': self})
-        sel.click(details_page.infoblock.element("Relationships", "Resources"))
-
     def wait_for_delete(self):
-        sel.force_navigate("clouds_stacks")
-        quad = Quadicon(self.name, 'stack')
-        wait_for(lambda: not sel.is_displayed(quad), fail_condition=False,
-            message="Wait stack to disappear", num_sec=500, fail_func=sel.refresh)
+        navigate_to(self, 'All')
+        wait_for(lambda: not sel.is_displayed(Quadicon(self.name, self.quad_name)),
+            fail_condition=False, message="Wait stack to disappear", num_sec=500,
+            fail_func=sel.refresh)
 
     def wait_for_appear(self):
-        sel.force_navigate("clouds_stacks")
-        quad = Quadicon(self.name, 'stack')
-        wait_for(sel.is_displayed, func_args=[quad], fail_condition=False,
-            message="Wait stack to appear", num_sec=1000, fail_func=sel.refresh)
-
-    def retire_stack(self):
-        sel.force_navigate('clouds_stack', context={'stack': self})
+        navigate_to(self, 'All')
+        wait_for(lambda: sel.is_displayed(Quadicon(self.name, self.quad_name)),
```

Comment:
I don't quite follow how this would change, maybe we can discuss it in chat. If you don't intend to have me implement in this PR go ahead and write the card/issue.